hipFuncAttributes struct definition needs to be C compliant

In `include/hip/hcc_detail/hip_runtime_api.h`:
`struct hipFuncAttributes { ... };` leads to error while compiling with gcc: `/opt/rocm/hip/include/hip/hcc_detail/hip_runtime_api.h:2268:33: error: unknown type name 'hipFuncAttributes'`

Changing it to `typedef struct hipFuncAttributes { ... } hipFuncAttributes;` seems to work.

--- a/include/hip/hcc_detail/hip_runtime_api.h
+++ b/include/hip/hcc_detail/hip_runtime_api.h
@@ -94,7 +94,7 @@ typedef struct ihipModule_t* hipModule_t;
 
 typedef struct ihipModuleSymbol_t* hipFunction_t;
 
-struct hipFuncAttributes {
+typedef struct hipFuncAttributes {
     int binaryVersion;
     int cacheModeCA;
     size_t constSizeBytes;
@@ -105,7 +105,7 @@ struct hipFuncAttributes {
     int preferredShmemCarveout;
     int ptxVersion;
     size_t sharedSizeBytes;
-};
+} hipFuncAttributes;
 
 typedef struct ihipEvent_t* hipEvent_t;
 

